### PR TITLE
Get rid of expects

### LINKED ISSUE #352: Get rid of expects
https://github.com/foundry-rs/starknet-foundry/pull/340#discussion_r1279265345

### PR DESCRIPTION
**Pull request by drknzz**
Closes #352 

## Introduced changes

- 
- 

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cast/src/starknet_commands/declare.rs [hunk 1]
```diff
@@ -116,15 +111,12 @@ pub async fn declare(
...
-            return Err(anyhow!(
-                "Failed to read starknet_artifacts.json contents: {}",
-                err
-            ))
-        }
-    };
+    let starknet_artifacts = std::fs::read_to_string(&starknet_artifacts).with_context(|| {
+        format!(
+            "Failed to read {} contents",
+            starknet_artifacts.into_os_string().into_string().unwrap()
```
Comment (THenry14) [inline crates/cast/src/starknet_commands/declare.rs:117]: not sure we should use unwrap here, what if this fails?

Comment (drknzz) [reply crates/cast/src/starknet_commands/declare.rs:117]: Would doing this be okay?
```rust
            "Failed to read {:?} contents",
            starknet_artifacts
```

File: crates/cast/src/helpers/scarb_utils.rs [hunk 2]
```diff
@@ -66,6 +66,17 @@ pub fn get_scarb_manifest() -> Result<Utf8PathBuf> {
...
 }
 
+pub fn get_scarb_metadata(manifest_path: &Utf8PathBuf) -> Result<scarb_metadata::Metadata> {
+    scarb_metadata::MetadataCommand::new()
+        .inherit_stderr()
+        .manifest_path(manifest_path)
+        .no_deps()
+        .exec()
+        .context(
+            "Failed to read Scarb.toml manifest file, not found in current nor parent directories",
```
Comment (war-in) [inline crates/cast/src/helpers/scarb_utils.rs:76]: Do we have a test for that? 🤔

File: crates/cast/src/starknet_commands/declare.rs (hunk 1 above)
Comment (THenry14) [reply crates/cast/src/starknet_commands/declare.rs:117]: yes, I think this is acceptable :+1:

**Review by war-in — COMMENTED** [review]
I wonder if those contexts will be displayed as follows

```
command: ...
error: ...
```

And if so, we should include some tests

**Review by war-in — APPROVED** [review]

**Pull request merged**